### ISSUE TITLE
[Accessibility][Text][Sample Content] Table headers not identified 

### ISSUE THREAD

**Issue by LevelAccessA11y** [adobe/aem-core-wcm-components#1675]
**Steps to Reproduce**
1. With NVDA operating, navigate sample table using Ctrl + Alt + arrow keys. When focus is on a table cell, press Ctrl + Alt + right arrow key, identify if the column header is announced. (Note: the default announcement for a table cell with no programmatically associated header should be "Column [n], [cell contents]". The text of the visual headers in the demo table match that default announcement. To effectively execute this test, change the first column header text to something distinguishable, e.g. "Test", then navigate to a cell in the second column and return to the first. If the cells are correctly associated, the announcement should be "Test, [cell contents if any]". Instead, it will be "Column 1, [cell contents if any]".)

**Current Behavior**
Table headers are not identified. Information that might be implied visually is not conveyed to screen reader users, which makes it difficult for these users to understand the structure of the table while navigating it.

**Expected behavior**
Ensure that table headers are programmatically identified. For simple tables, headers should be identified with the `<th>` element. 

**Environment**
Chrome 91, Windows 10, NVDA 2020.4 
AEM Core Components v. 2.17  

**Affected Users**
Assistive technology users

**WCAG SC**
1.3.1 Info and Relationships

**Code snippet**
```
<table border="1" cellspacing="0" cellpadding="5">
    <tbody>
        <tr>
            <td>Column 1</td>
            <td>Column 2</td>
            <td>Column 3</td>
            <td>Column 4</td>
        </tr>
        <tr>
            <td></td>
            ...
        </tr>
        ...
    </tbody>
</table>
```

**Comment by bpauli** [comment]
@adobe export issue to Jira project SITES

**Comment by gabrielwalt** [comment]
This is only a content issue for the sample content used by the Component Library on following page:
https://www.aemcomponents.dev/content/core-components-examples/library/page-authoring/text.html
